addons/mail/static/src/models/notification_group/notification_group.js

```diff
@@ -67,6 +67,28 @@ function factory(dependencies) {
             return `${this.modelName}_${data.id}`;
         }
 
+        _computeDate() {
+            const groupDate = moment.max(this.notifications.map(notification => notification.message.date));
+            if (groupDate.isValid()) {
```

Comment:
This check is confusing, when it is supposed to become invalid?
It would be better to handle bad cases upfront (not giving to moment empty dates, or whatever is the situation).